### PR TITLE
[DOCS] Adds cat datafeeds API

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/issues/51411

This PR adds documentation for the cat datafeeds API.

Preview: http://elasticsearch_52738.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/cat-datafeeds.html

### CODE REVIEW

**Comment by lcawl** [comment]
@elasticmachine update branch

File: docs/reference/cat/datafeeds.asciidoc [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
++
+include::{docdir}/ml/ml-shared.asciidoc[tag=node]
++
+To retrieve this information, specify the `nn` or `nodeName` column in the `h`
+query parameter.
+
+`search.bucket_avg`::
+include::{docdir}/ml/ml-shared.asciidoc[tag=search-bucket-avg]
++
+To retrieve this information, specify the `sba` or `bucketTimeMin` column in the
```
Comment (benwtrent) [inline docs/reference/cat/datafeeds.asciidoc]: this is wrong. The API is also wrong as well. I will correct in a PR. Then we can correct it here :)

Comment (benwtrent) [reply docs/reference/cat/datafeeds.asciidoc]: This should be `searchBucketAvg`. Could you change it here and in https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/cat/RestCatDatafeedsAction.java#L95

Comment (lcawl) [reply docs/reference/cat/datafeeds.asciidoc]: Good catch, thanks @benwtrent !  I've made that change

**Review by benwtrent — APPROVED** [review]

**Pull request merged**